frontends/common/options.cpp

```diff
@@ -130,6 +133,38 @@ void CompilerOptions::setInputFile() {
     }
 }
 
+std::vector<const char*>* CompilerOptions::process(int argc, char* const argv[]) {
+    char buffer[PATH_MAX];
+    int len;
+    struct stat st;
+    if ((len = readlink("/proc/self/exe", buffer, sizeof(buffer))) > 0 ||
```

Comment:
dirname(argv[0]) is the fallback and only works if the program was invoked with an absolute path.  We use cwd+dirname(argv[0]) for a relative path.  We could add searching through the $PATH envvar for cases where there is no path in argv[0].